src/pageTemplates/HistoryTemplate/index.tsx

```diff
@@ -1,27 +1,29 @@
-import { fetchHistory } from '@/api/history';
-import { Heading } from '@/components/Heading';
-import { LayoutWithSidebar } from '@/components/layouts/LayoutWithSidebar';
-import { Paragraph } from '@/components/Paragraph';
-import { Table } from '@/components/Table';
-import { MOCK_HISTORY } from '@/constants/history';
-import { handleToast } from '@/utils/handleToast';
-import { useCallback, useEffect, useState } from 'react';
+import { fetchHistory } from "@/api/history";
+import { Heading } from "@/components/Heading";
+import { LayoutWithSidebar } from "@/components/layouts/LayoutWithSidebar";
+import { Paragraph } from "@/components/Paragraph";
+import { Table } from "@/components/Table";
+import { handleToast } from "@/utils/handleToast";
+import { useCallback, useEffect, useState } from "react";
 
 export const HistoryTemplate = () => {
-  const [history, setHistory] = useState(MOCK_HISTORY);
+  const [history, setHistory] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
 
-  const handleFetchHistory = useCallback(async (search?: string) => {
-    setIsLoading(true);
-    try {
-      const { history: data } = await fetchHistory(search ?? '');
-      setHistory(data);
-    } catch (err) {
-      handleToast('Algo deu errado.', 'error');
-    } finally {
-      setIsLoading(false);
-    }
-  }, []);
+  const handleFetchHistory = useCallback(
+    async (search?: string, page?: number) => {
+      setIsLoading(true);
+      try {
+        const { history: data } = await fetchHistory(search ?? "", page ?? 1);
```

Comment:
No `fetchHistory()`, você recebe uma propriedade chamada `totalPages`. Use isso para a paginação.
Passe o `totalPages` como propriedade pra table, receba no componente Table.
Com o `totalPages` dentro de table, não deixe o usuário paginar para uma página que existe. No `nextStep()` faça uma verificação: se a página que o usuário tiver tentando acessar for maior que o totalPages, **não permita**.

Reply:
Todas as páginas que usamos o fetch(), HistoryTemplate, InventoryTemplate, etc... Usam paginação. Aplique pra o restante das páginas tb.